src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/transport/action/updateindexmetadata/UpdateManagedIndexMetaDataRequest.kt

```diff
@@ -24,38 +24,56 @@ import org.elasticsearch.common.io.stream.StreamOutput
 import org.elasticsearch.index.Index
 
 class UpdateManagedIndexMetaDataRequest : AcknowledgedRequest<UpdateManagedIndexMetaDataRequest> {
-    lateinit var listOfIndexMetadata: List<Pair<Index, ManagedIndexMetaData>>
+
+    lateinit var indicesToAddManagedIndexMetaDataTo: List<Pair<Index, ManagedIndexMetaData>>
+        private set
+
+    lateinit var indicesToRemoveManagedIndexMetaDataFrom: List<Index>
         private set
 
     constructor()
 
-    constructor(listOfIndexMetadata: List<Pair<Index, ManagedIndexMetaData>>) {
-        this.listOfIndexMetadata = listOfIndexMetadata
+    constructor(
+        indicesToAddManagedIndexMetaDataTo: List<Pair<Index, ManagedIndexMetaData>> = listOf(),
+        indicesToRemoveManagedIndexMetaDataFrom: List<Index> = listOf()
+    ) {
+        this.indicesToAddManagedIndexMetaDataTo = indicesToAddManagedIndexMetaDataTo
+        this.indicesToRemoveManagedIndexMetaDataFrom = indicesToRemoveManagedIndexMetaDataFrom
     }
 
     override fun validate(): ActionRequestValidationException? {
         var validationException: ActionRequestValidationException? = null
-        if (!this::listOfIndexMetadata.isInitialized) {
-            validationException = addValidationError("must specify index List for UpdateManagedIndexMetaData", validationException)
+        if (!this::indicesToAddManagedIndexMetaDataTo.isInitialized && !this::indicesToRemoveManagedIndexMetaDataFrom.isInitialized) {
```

Comment:
I think this is not really necessary now. Previously there was no default list and the list always came from the constructor. Now that is not the case and we have default empty `listOf`. We should instead check if both of them are empty list then throw an exception .

Reply:
Added a check to ensure that at least one non-empty list is given but will also keep the check to validate that both lists are initialized for robustness.